src/hyper.rs

```diff
@@ -0,0 +1,236 @@
+use http_types::headers::{HeaderName, HeaderValue};
+use http_types::{Method, StatusCode, Version};
+use hyper::body::HttpBody;
+use hyper_tls::HttpsConnector;
+use std::convert::{TryFrom, TryInto};
+use std::str::FromStr;
+
+/// Hyper-based HTTP Client.
+#[derive(Debug)]
+pub struct HyperClient {}
+
+impl HyperClient {
+    /// Create a new client.
+    ///
+    /// There is no specific benefit to reusing instances of this client.
+    pub fn new() -> Self {
+        HyperClient {}
+    }
+}
+
+impl HttpClient for HyperClient {
+    fn send(&self, req: Request) -> futures::future::BoxFuture<'static, Result<Response, Error>> {
+        Box::pin(async move {
+            let req = HyperHttpRequest::try_from(req).await?.into_inner();
+            // UNWRAP: Scheme guaranteed to be "http" or "https" as part of conversion
+            let scheme = req.uri().scheme_str().unwrap();
+
+            let response = match scheme {
+                "http" => {
+                    let client = hyper::Client::builder().build_http::<hyper::Body>();
+                    client.request(req).await
+                }
+                "https" => {
+                    let https = HttpsConnector::new();
+                    let client = hyper::Client::builder().build::<_, hyper::Body>(https);
+                    client.request(req).await
+                }
+                _ => unreachable!(),
+            }?;
+
+            let resp = HttpTypesResponse::try_from(response).await?.into_inner();
+            Ok(resp)
+        })
+    }
+}
+
+struct HyperHttpRequest {
+    inner: hyper::Request<hyper::Body>,
+}
+
+impl HyperHttpRequest {
+    async fn try_from(mut value: Request) -> Result<Self, Error> {
+        let method = match value.method() {
+            Method::Get => hyper::Method::GET,
+            Method::Head => hyper::Method::HEAD,
+            Method::Post => hyper::Method::POST,
+            Method::Put => hyper::Method::PUT,
+            Method::Patch => hyper::Method::PATCH,
+            Method::Options => hyper::Method::OPTIONS,
+            Method::Trace => hyper::Method::TRACE,
+            Method::Connect => hyper::Method::CONNECT,
+            _ => {
+                return Err(Error::from_str(
+                    StatusCode::BadRequest,
+                    "unrecognized HTTP method",
+                ))
+            }
+        };
+
+        let version = value
+            .version()
+            .map(|v| match v {
+                Version::Http0_9 => Ok(hyper::Version::HTTP_09),
+                Version::Http1_0 => Ok(hyper::Version::HTTP_10),
+                Version::Http1_1 => Ok(hyper::Version::HTTP_11),
+                Version::Http2_0 => Ok(hyper::Version::HTTP_2),
+                Version::Http3_0 => Ok(hyper::Version::HTTP_3),
+                _ => Err(Error::from_str(
+                    StatusCode::BadRequest,
+                    "unrecognized HTTP version",
+                )),
```

Comment:
`http-types` has a [`hyperium-http` feature](https://github.com/http-rs/http-types/blob/master/Cargo.toml#L22) that implements these conversions for you.

Reply:
Fixed!